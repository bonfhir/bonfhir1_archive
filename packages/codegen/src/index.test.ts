import { exec } from "node:child_process";
import { readFile } from "node:fs/promises";
import { join } from "node:path";
import { promisify } from "node:util";

const execAsync = promisify(exec);

describe("codegen", () => {
  describe("run", () => {
    it("process data and templates", async () => {
      const cli = "yarn dev";
      const data = join(__dirname, "__fixtures__", "data", "*.json");
      const templates = join(__dirname, "__fixtures__", "templates", "*.hbs");
      const helper = join(
        __dirname,
        "__fixtures__",
        "templates",
        "test-helper.js"
      );
      await execAsync(
        `${cli} run -d ${data} -t ${templates} --helpers "${helper}" -p "prettier --write %files%"`,
        {
          cwd: __dirname,
        }
      );

      const loadedResult = await readFile(
        join(__dirname, "__fixtures__", "templates", "result.ts"),
        "utf8"
      );
      expect(loadedResult).toMatchSnapshot();
    });
  });
});
