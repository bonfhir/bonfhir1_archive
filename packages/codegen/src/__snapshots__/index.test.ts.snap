// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codegen run process data and templates 1`] = `
"export const AllDomainResources = [
  "prefixAccount",
  "prefixActivityDefinition",
  "prefixAdministrableProductDefinition",
  "prefixAdverseEvent",
  "prefixAllergyIntolerance",
  "prefixAuditEvent",
  "prefixBasic",
  "prefixBiologicallyDerivedProduct",
  "prefixBodyStructure",
  "prefixCapabilityStatement",
  "prefixCarePlan",
  "prefixCareTeam",
  "prefixCatalogEntry",
  "prefixChargeItem",
  "prefixChargeItemDefinition",
  "prefixCitation",
  "prefixClaim",
  "prefixClaimResponse",
  "prefixClinicalImpression",
  "prefixClinicalUseDefinition",
  "prefixCodeSystem",
  "prefixCommunication",
  "prefixCommunicationRequest",
  "prefixCompartmentDefinition",
  "prefixComposition",
  "prefixConceptMap",
  "prefixCondition",
  "prefixConsent",
  "prefixContract",
  "prefixCoverage",
  "prefixCoverageEligibilityRequest",
  "prefixCoverageEligibilityResponse",
  "prefixDetectedIssue",
  "prefixDevice",
  "prefixDeviceDefinition",
  "prefixDeviceMetric",
  "prefixDeviceRequest",
  "prefixDeviceUseStatement",
  "prefixDiagnosticReport",
  "prefixDocumentManifest",
  "prefixDocumentReference",
  "prefixEncounter",
  "prefixEndpoint",
  "prefixEnrollmentRequest",
  "prefixEnrollmentResponse",
  "prefixEpisodeOfCare",
  "prefixEventDefinition",
  "prefixEvidence",
  "prefixEvidenceReport",
  "prefixEvidenceVariable",
  "prefixExampleScenario",
  "prefixExplanationOfBenefit",
  "prefixFamilyMemberHistory",
  "prefixFlag",
  "prefixGoal",
  "prefixGraphDefinition",
  "prefixGroup",
  "prefixGuidanceResponse",
  "prefixHealthcareService",
  "prefixImagingStudy",
  "prefixImmunization",
  "prefixImmunizationEvaluation",
  "prefixImmunizationRecommendation",
  "prefixImplementationGuide",
  "prefixIngredient",
  "prefixInsurancePlan",
  "prefixInvoice",
  "prefixLibrary",
  "prefixLinkage",
  "prefixList",
  "prefixLocation",
  "prefixManufacturedItemDefinition",
  "prefixMeasure",
  "prefixMeasureReport",
  "prefixMedia",
  "prefixMedication",
  "prefixMedicationAdministration",
  "prefixMedicationDispense",
  "prefixMedicationKnowledge",
  "prefixMedicationRequest",
  "prefixMedicationStatement",
  "prefixMedicinalProductDefinition",
  "prefixMessageDefinition",
  "prefixMessageHeader",
  "prefixMolecularSequence",
  "prefixNamingSystem",
  "prefixNutritionOrder",
  "prefixNutritionProduct",
  "prefixObservation",
  "prefixObservationDefinition",
  "prefixOperationDefinition",
  "prefixOperationOutcome",
  "prefixOrganization",
  "prefixOrganizationAffiliation",
  "prefixPackagedProductDefinition",
  "prefixPatient",
  "prefixPaymentNotice",
  "prefixPaymentReconciliation",
  "prefixPerson",
  "prefixPlanDefinition",
  "prefixPractitioner",
  "prefixPractitionerRole",
  "prefixProcedure",
  "prefixProvenance",
  "prefixQuestionnaire",
  "prefixQuestionnaireResponse",
  "prefixRegulatedAuthorization",
  "prefixRelatedPerson",
  "prefixRequestGroup",
  "prefixResearchDefinition",
  "prefixResearchElementDefinition",
  "prefixResearchStudy",
  "prefixResearchSubject",
  "prefixRiskAssessment",
  "prefixSchedule",
  "prefixSearchParameter",
  "prefixServiceRequest",
  "prefixSlot",
  "prefixSpecimen",
  "prefixSpecimenDefinition",
  "prefixStructureDefinition",
  "prefixStructureMap",
  "prefixSubscription",
  "prefixSubscriptionStatus",
  "prefixSubscriptionTopic",
  "prefixSubstance",
  "prefixSubstanceDefinition",
  "prefixSupplyDelivery",
  "prefixSupplyRequest",
  "prefixTask",
  "prefixTerminologyCapabilities",
  "prefixTestReport",
  "prefixTestScript",
  "prefixValueSet",
  "prefixVerificationResult",
  "prefixVisionPrescription",
];
"
`;
